stocks2.lua

```diff
@@ -0,0 +1,544 @@
+                        {text = "Shift-W", pen = COLOR_LIGHTGREEN}, {text = ": Min Wear: ", pen = COLOR_WHITE}, {text = "tbd", pen = COLOR_BROWN}
+                    },
+                    frame = {t = 12, l = 0},
+                    view_id = "filters_qual_wear",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Actions (", pen = COLOR_BROWN}, {text = "...", pen = COLOR_LIGHTGREEN}, {text = " Items)", pen = COLOR_BROWN}
+                    },
+                    frame = {t = 16, l = 0},
+                    view_id = "actions_item_count_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-Z", pen = COLOR_LIGHTGREEN}, {text = ": Zoom    ", pen = COLOR_WHITE},
+                        {text = "-D", pen = COLOR_LIGHTGREEN}, {text = ": Dump", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-F", pen = COLOR_LIGHTGREEN}, {text = ": Forbid  ", pen = COLOR_WHITE},
+                        {text = "-M", pen = COLOR_LIGHTGREEN}, {text = ": Melt", pen = COLOR_WHITE}, NEWLINE,
+                        {text = "Shift-T", pen = COLOR_LIGHTGREEN}, {text = ": Mark for Trade", pen = COLOR_GREY},
+                    },
+                    frame = {t = 17, l = 0},
+                    view_id = "actions_label",
+                }),
+                widgets.Label({
+                    text = {
+                        {text = "Shift-A", pen = COLOR_LIGHTGREEN}, {text = ": Apply to: ", pen = COLOR_WHITE},
+                        {text = apply_to_labels[apply_to_mode + 1], pen = COLOR_BROWN},
+                    },
+                    frame = {t = 20, l = 0},
+                    view_id = "apply_to_label",
+                }),
+            }
+        }),
+    })
+
+    self:populate_item_table()
+    self:filter_list(current_search_query)
+    self:show_filters()
+    self:show_qual_and_wear()
+end
+
+function StocksWindow:populate_item_table()
+    -- retrieve all in-play items
+    local temp_table = {}
+    for i,v in ipairs(df.global.world.items.other[df.items_other_id.IN_PLAY]) do
+        local item_desc = dfhack.items.getDescription(v, v:getType())
+
+        -- store item description and an item ref
+        table.insert(temp_table, { desc = item_desc, ref = v })
+    end
+
+    -- sort alphabetically
+    local spec = { key = function(v) return v.desc end }
+    local order = utils.make_sort_order(temp_table, { spec })
+    for i = 1, #order do item_table[i] = temp_table[order[i]] end
+end
+
+function toggle_melt(item)
+    item.flags.melt = not item.flags.melt
+
+    if item.flags.melt then
+        utils.insert_sorted(df.global.world.items.other.ANY_MELT_DESIGNATED, item, "id")
+    else
+        utils.erase_sorted(df.global.world.items.other.ANY_MELT_DESIGNATED, item, "id")
+    end
+end
+
+function is_valid_melt_item(item)
```

Comment:
this should be called from the logistics plugin instead of rewritten here. Could you open another PR to make the the logic at https://github.com/DFHack/dfhack/blob/develop/plugins/logistics.cpp#L269 callable from Lua? This just means move that logic to a static function and add it to `DFHACK_PLUGIN_LUA_FUNCTIONS` here: https://github.com/DFHack/dfhack/blob/develop/plugins/logistics.cpp#L744
then you can add `local logistics = require('plugins.logistics') to the top of this file and call it via `logistics.logistics_is_meltable_item(item)`